share/doc/wake/logging.md

```diff
@@ -0,0 +1,152 @@
+each logger's output will be directed to zero or more of wake's outputs streams.
+
+For `wake`'s own stdout, the logger's color & intensity are used to apply
+additional formatting to the output before displaying it.
+
+For understanding which streams will appear on wake's stdout,
+the following table should be read top to bottom considering the command line arguments you've passed to wake.
+If you match a line in the table, stop!
+This is what will appear on the console as you watch wake execute,
+i.e. what is sent to ``wake`'s stdout:
+
+| log level           | "debug"  | "info"  | "echo"  | "report"  | "warning"  | "error"  | "null"  |  "foo" |
+|---------------------|----------|---------|---------|-----------|------------|----------|---------|--------|
+|--stdout="foo"       |          |         |         |           |            |          |         |    x   |
+|--stdout="foo,info"  |          |    x    |         |           |            |          |         |    x   |
+|--debug              |      x   |    x    |    x    |     x     |      x     |    x     |         |        |
+|--verbose            |          |    x    |    x    |     x     |      x     |    x     |         |        |
+|--quiet              |          |         |         |           |            |    x     |         |        |
+|--no-tty<sup>*</sup> |          |         |    x    |     x     |      x     |    x     |         |        |
+|(default)            |          |         |         |     x     |      x     |    x     |         |        |
+
+<sup>*</sup> `--no-tty` also prevents any coloring from being applied.
+
+Unless overridden with `--stderr=...` the *only* thing that ever appears on wake's stderr is outputs from loggers
+with log level "error" (e.g. `logError`).
+
+The additional command line arguments `--fd:3`, `--fd:4`, `--fd:5` can be used to output to file descriptors 3, 4, 5.
+One example shell command would be, for some wake code that had `mkLogLevel "foo"`:
+
+```
+wake --fd:3="foo" mywakecode 3>foo.txt
+```
+
+These can be concatenated with commas and include the normal log levels, so to capture both `mkLogLevel "foo"` and `logInfo` to a file:
+
+```
+wake --fd:3="foo,info" mywakecode 3>foo_and_info.txt
+```
+
+## What Gets Sent To Loggers
+
+### Job Execution
+
+Every `Job` run by wake tracks its own stdout and stderr,
+and when the underlying process prints to stdout or stderr,
+it will go to the `Job`-specific stdout and stderr,
+not the stdout and stderr of the `wake` process itself.
+
+When writing your wake code,
+you can control which loggers these `Job`-specific stdout and stderr are sent to.
+This will influence whether they appear on the `wake` process's own stdout,
+stderr, or other arbitrary file descriptors.
+These can be controlled with `setPlanStdout`, `setPlanStderr` before you `runJob`.
+These default to `logInfo` and `logError` respectively.
+
+Regardless of to which logger you direct the `Job`'s `Stderr` and `Stdout`,
```

Comment:
This is perfectly valid grammar, but the double preposition feels like it stumbles a bit on the first reading.  Maybe rearrange the "to"?
```suggestion
Regardless of the logger to which you direct the `Job`'s `Stderr` and `Stdout`,
```